app/carbonapi/http_handlers.go

```diff
@@ -993,11 +993,11 @@ func blockHeaders(w http.ResponseWriter, r *http.Request) {
 	if len(m) == 0 {
 		logger.Error("couldn't create a rule from params")
 	} else {
-		fileData, err := loadBlockRuleConfig()
+		fileData, err := loadBlockRuleConfig(app)
 		if err == nil {
 			yaml.Unmarshal(fileData, &ruleConfig)
 		}
-		err1 = appendRuleToConfig(ruleConfig, m, logger)
+		err1 = appendRuleToConfig(ruleConfig, m, logger, app)
```

Comment:
I think it'd be a bit better to just pass in the `app.config.BlockHeaderFile` here, since that appears to be the only field used under the `appendRuleToConfig` call tree.